zerver/lib/bugdown/__init__.py

```diff
@@ -1156,18 +1148,18 @@ def run(self, root: Element) -> None:
             if not self.markdown.url_embed_preview_enabled:
                 continue
 
+            if youtube is not None:
+                if youtube['title'] is not None:
+                    found_url.family.child.text = youtube['title']
+                continue
+
```

Comment:
Reasons behind moving this piece of code here are;
* We don't use to extracted data from cache anywhere in case of YouTube embeds so `extracted_data = link_preview.link_embed_data_from_cache(url)` is redundant here.

* It's simpler to change the test function `test_youtube_url_title_replaces_url` because we won't need to mock `link_embed_data_from_cache` calls.